File: bro-agent-api/src/main/python/bro_agent/agent/RestoreState.py
from abc import ABC
import bro_agent.agent.states

from bro_agent.generated.Action_pb2 import Action


class RestoreState(bro_agent.agent.AgentState.AgentState, ABC):
    """ Represents agent participating in a restore. """

    def __init__(self, agent, restore_information):
        # @param agent The agent performing the restore
        # @param restoreInformation information required for restore
        bro_agent.agent.AgentState.AgentState.__init__(self, agent)
        self.restore_information = restore_information

    def cancel_action(self):
        """ Canceling the Action """
        return bro_agent.agent.CancelActionState.CancelActionState(self.agent,
                                                                   self.restore_information.backup_name,
                                                                   Action.RESTORE)

File: bro-agent-api/src/main/java/com/ericsson/adp/mgmt/bro/api/agent/RestoreState.java
/*------------------------------------------------------------------------------
 *******************************************************************************
 * COPYRIGHT Ericsson 2019
 *
 * The copyright to the computer program(s) herein is the property of
 * Ericsson Inc. The programs may be used and/or copied only with written
 * permission from Ericsson Inc. or in accordance with the terms and
 * conditions stipulated in the agreement/contract under which the
 * program(s) have been supplied.
 *******************************************************************************
 *----------------------------------------------------------------------------*/
package com.ericsson.adp.mgmt.bro.api.agent;

import com.ericsson.adp.mgmt.action.Action;

/**
 * Represents agent participating in a restore.
 */
public abstract class RestoreState extends AgentState {

    protected final RestoreInformation restoreInformation;

    /**
     * @param agent The agent performing the restore
     * @param restoreInformation information required for restore
     */
    public RestoreState(final Agent agent, final RestoreInformation restoreInformation) {
        super(agent);
        this.restoreInformation = restoreInformation;
    }

    @Override
    protected AgentState cancelAction() {
        return new CancelActionState(this.agent, this.restoreInformation.getBackupName(), Action.RESTORE);
    }

}
